### PR TITLE
common: add support for new Sepolia chain

### PR DESCRIPTION
**Pull request by emersonmacro**
Initial attempt at tackling #1558. Information for the new chain came primarily from the geth issue & PR here:
https://github.com/ethereum/go-ethereum/pull/23730#issuecomment-942646618
https://github.com/ethereum/go-ethereum/pull/23730/files#diff-c13a1b5c84048a832e07e3503cc35f66614816cb39fe1e2fdd992672a9e1d883R46

### CODE REVIEW

File: packages/common/src/chains/sepolia.json [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  "name": "sepolia",
+  "chainId": 11155111,
+  "networkId": 11155111,
+  "defaultHardfork": "istanbul",
+  "consensus": {
+    "type": "pow",
+    "algorithm": "ethash",
+    "ethash": {}
+  },
+  "comment": "PoW test network",
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:11]: Something more specific would be nice here "PoW test network to replace Ropsten" e.g..

File: packages/common/src/chains/sepolia.json [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  "url": "https://github.com/ethereum/ropsten",
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:12]: Please update this URL with something fitting, a robust PR URL like https://github.com/ethereum/go-ethereum/pull/23730 would also be ok (this info is not read "by machine" anywhere, just some helper for someone digging into the chain files).

File: packages/common/src/chains/sepolia.json [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  "genesis": {
+    "hash": "0x25a5cc106eea7138acab33231d7160d69cb777ee0c2c553fcddf5138993e6dd9",
+    "timestamp": "0x6159af19",
+    "gasLimit": 30000000,
+    "difficulty": 131072,
+    "nonce": "0x000000000000000",
+    "extraData": "0x5365706f6c69612c20417468656e732c204174746963612c2047726565636521",
+    "stateRoot": "0x5eb6e371a698b8d68f665192350ffcecbbbf322916f4b51bd79bb6887da3f494"
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:20]: Did you took over `hash` and `stateRoot` here from somewhere or did you calculate yourself?

Can you confirm the correctness of both?

**Review by holgerd77 — COMMENTED** [review]
Review is not yet complete (to be continued by me or someone else).

Is this Sepolia network already running? If so: have you (@emersonmacro) tested this setup in the client?

File: packages/common/src/chains/sepolia.json (hunk 3 above)
Comment (acolytec3) [reply packages/common/src/chains/sepolia.json:20]: I'll need to compare in more detail but here's what Besu is using for Genesis config.https://github.com/hyperledger/besu/blob/main/config/src/main/resources/sepolia.json

Comment (emersonmacro) [reply packages/common/src/chains/sepolia.json:20]: I got them from this comment on the original geth PR - https://github.com/ethereum/go-ethereum/pull/23730#issuecomment-942646618. I'll see if I can confirm them with our client

Comment (holgerd77) [reply packages/common/src/chains/sepolia.json:20]: Ah, ok, but that's also already a solid source. 👍

File: packages/common/src/chains/sepolia.json [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    }, 
+    {
+      "name": "london",
+      "block": 0, 
+      "forkHash": "0x30c7ddbc"
+    },
+    {
+      "name": "arrowGlacier",
+      "block": 0,
+      "forkHash": "0x30c7ddbc"
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:81]: If I am not missing something, the `arrowGlacier` HF here should not be included respectively is not applied on `sepolia` yet. I guess `arrowGlacier` will be applied at some point (?) but maybe for now remove the entry completely.

File: packages/common/src/chains/sepolia.json [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    {
+      "name": "tangerineWhistle",
+      "block": 0,
+      "forkHash": "0x30c7ddbc"
+    },
+    {
+      "name": "spuriousDragon",
+      "block": 0,
+      "forkHash": "0x30c7ddbc"
+    },
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:42]: The `arrowGlacier` removal now also changes the fork hash (on all places), this is now `0xfe3366e7`.

File: packages/common/src/chains/sepolia.json [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    "ethash": {}
+  },
+  "comment": "PoW test network to replace Ropsten",
+  "url": "https://github.com/ethereum/go-ethereum/pull/23730",
+  "genesis": {
+    "hash": "0x25a5cc106eea7138acab33231d7160d69cb777ee0c2c553fcddf5138993e6dd9",
+    "timestamp": "0x6159af19",
+    "gasLimit": 30000000,
+    "difficulty": 131072,
+    "nonce": "0x000000000000000",
```
Comment (holgerd77) [inline packages/common/src/chains/sepolia.json:18]: The nonce is off by one 0.

**Review by holgerd77 — CHANGES_REQUESTED** [review]
Some change requests, I will directly push these after the review.

The initial connection to the bootnode is now working with this. 

We get the following follow-up error though:

grafik

Guess that has something to do with the genesis block which might not be written properly to the chain. This will need some debugging. A good starting point here is likely the blockchain library and there the related genesis block initialization code parts.

**Comment by holgerd77** [comment]
What's the status of this?

**Comment by emersonmacro** [comment]
> What's the status of this?

I was able to get the same initial connection to the bootnode and subsequent error that you posted. I'm doing some debugging trying to figure out what's causing the error.

**Comment by ryanio** [comment]
sync working! smooth so far

Screen Shot 2021-12-30 at 10 14 55 AM

**Comment by ryanio** [comment]
i successfully sync'd to head but not sure why it didn't continue following the tip of the chain

Screen Shot 2021-12-30 at 11 17 02 AM

This error:
> `(node:24407) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [BlockFetcher]. Use emitter.setMaxListeners() to increase limit`

is concerning, might be related to `lib/sync/fetcher/fetcher.ts:L348` -> `this.on('close', () => {`

**Comment by jochem-brouwer** [comment]
Maybe no one is mining Sepolia so there are no new blocks? Check the timestamp of the latest block?

**Comment by holgerd77** [comment]
But this Sepolia network is a great chance to really really test our client in production, and all the related tip of the chain behavior with it! 😀

**Comment by emersonmacro** [comment]
I got it fully synced yesterday and did not see the same error:
image

Just restarted the client and it synced back up very quickly, at a slightly higher block height:
image

**Review by ryanio — APPROVED** [review]
this LGTM!

we can address following the tip of chain behavior in a separate PR, and this should be a nice testnet to do that with :)

**Review by ryanio — APPROVED** [review]

**Pull request merged**